lib/benchpark/system.py

```diff
@@ -43,6 +53,9 @@ def __init__(self, spec):
         self.timeout = "120"
         self.queue = None
 
+        # Assume every system is an MPI/OpenMP system
+        self._programming_models = [MPISystem(), OpenMPSystem()]
```

Comment:
@pearce8 I know we assume MPI, but is there an example of a system which **is not** also an `OpenMPSystem`?

Reply:
We might need 2 types of OpenMP systems:

- [ ] OpenMPSystem - with OpenMP that works on the CPU.  We should not make all systems by default OpenMPSystems because it depends on whether there is an OpenMP compiler.
- [ ] OpenMPOffloadSystem - if systems with GPUs have a compiler which is able to offload work to the GPU using an OpenMP directive.  Again, only add this if it indeed works - currently, we would only ever do that with Fortran, so we do not have any use cases for this.